src/utp_wrapper.rs

```diff
@@ -47,12 +48,20 @@ impl UtpWrapper {
                                     }
                                 },
                                 Err(TryRecvError::Disconnected) => break 'outer,
-                                Err(TryRecvError::Empty) => break,
+                                Err(TryRecvError::Empty) => {
+                                    attempts += 1;
+                                    if attempts < 5000 {
+                                        break;
+                                    } else {
+                                        break 'outer;
```

Comment:
no you don't have to. the loop is around try_recv , which means if there is nothing in the queue, it will `break` eventually.
However, it maybe better change to use a timeout, but that shall be after CI confirms the change works.